app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -175,7 +177,7 @@ open class HomeActivity : LocaleAwareAppCompatActivity(), NavHostActivity {
         setupThemeAndBrowsingMode(getModeFromIntentOrLastKnown(intent))
         setContentView(R.layout.activity_home)
 
-        createNavGraph()
+        val navGraphJob = createNavGraph()
```

Comment:
Can this break if the inflation takes too long? We can navigate using the `navController` before `onCreate` finishes. For example, see `it.process` (below) -> OpenBrowserIntentProcessor -> activity.openToBrowser which calls `navController.nav` (which I assume needs to have the navGraph available, but maybe not!). You can probably test this by calling `delay` in the async task before it assigns the navGraph.

If so, one thought: could we start async inflation in `FenixApplication`? e.g. get the navInflater (programmatically creating a NavHostFragment if we need to) and starting the inflation early?